fmriprep/utils/confounds.py

```diff
@@ -0,0 +1,138 @@
+"""Utilities for confounds manipulation."""
+
+
+def mask2vf(in_file, zooms=None, out_file=None):
+    """
+    Convert a binary mask on a volume fraction map.
+
+    The algorithm simply applies a Gaussian filter with the kernel size scaled
+    by the zooms given as argument.
+
+    """
+    import numpy as np
+    import nibabel as nb
+    from scipy.ndimage import gaussian_filter
+
+    img = nb.load(in_file)
+    imgzooms = np.array(img.header.get_zooms()[:3], dtype=float)
+    if zooms is None:
+        zooms = imgzooms
+
+    zooms = np.array(zooms, dtype=float)
+    sigma = 0.5 * (zooms / imgzooms)
+    if len(sigma) > 1:
+        sigma = tuple(sigma)
+
+    data = gaussian_filter(img.get_fdata(dtype=np.float32), sigma=sigma)
+
+    max_data = np.percentile(data[data > 0], 99)
+    data = np.clip(data / max_data, a_min=0, a_max=1)
+
+    if out_file is None:
+        return data
+
+    hdr = img.header.copy()
+    hdr.set_data_dtype(np.float32)
+    nb.Nifti1Image(data.astype(np.float32), img.affine, hdr).to_filename(out_file)
+    return out_file
+
+
+def acompcor_masks(in_files, is_aseg=False, zooms=None):
```

Comment:
I don't really like sticking major pieces of machinery into a `utils` module and having the interfaces be thin wrappers on a hidden function. I think at one point we were doing something like:

```Python
class AlgorithmInterface:
    def _run_interface(self, runtime):
        self._results = _algorithm(**self.inputs.get())
        return runtime

def _algorithm(arg1, arg2):
    ...
    return {"out1": ...}
```

And the justification was that it would be easier to test a function than an interface. We're not testing this, and now it's being hidden away into an unassuming corner of the code.

I don't want to hold up the PR, but I would suggest we think through our organization and apply it consistently.

Reply:
> I don't want to hold up the PR, but I would suggest we think through our organization and apply it consistently.

Yup, this would be good. Let's make sure we find a time after the LTS. At first my only comment would be that sphinx does not take private functions by default (that's why this is public). That said, if when we meet we decide to have everything together, fine by me - this was kind of an arbitrary decision.